pkgs/top-level/emscripten-packages.nix

```diff
@@ -0,0 +1,107 @@
+{ pkgs }:
+
+# emscripten toolchain abstraction for nix
+# https://github.com/NixOS/nixpkgs/pull/16208
+
+with pkgs; rec {
+  json_c = pkgs.json_c.override {
+    stdenv = emscriptenStdenv;
+  };
+  
+  libxml2 = (pkgs.libxml2.override {
+    stdenv = emscriptenStdenv;
+    supportPython = false;
+  }).overrideDerivation
+    (old: { 
+      buildInputs = old.buildInputs ++ [ autoreconfHook pkgconfig zlib nodejs ];
+      nativeBuildInputs = old.nativeBuildInputs ++ [ zlib pkgconfig ];
+      # just override it with nothing so it does not fail
+      autoreconfPhase = "echo autoreconfPhase not used..."; 
```

Comment:
took me already hours to figure what is going wrong. not sure what happens there but it works now -> so i won't change it anymore.
the `autoreconfHook` seems to be required in `libxml2`'s native application but is not usable in the emscripten world.